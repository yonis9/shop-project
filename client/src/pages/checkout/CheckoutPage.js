import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { 
    cartItemsTotalSelector,
    cartItemsSelector } from '../../redux/cart/cart-selectors'

import CheckoutItem from '../../components/CheckoutItem/CheckoutItem';
import StripeButton from '../../components/StripeButton/StripeButton';

import {
    CheckoutPageContainer,
    CheckoutHeaderContainer,
    HeaderBlockContainer,
    TotalContainer,
    TestWarningContainer
} from './CheckoutPage.styles';

export const CheckoutPage = ({ cartItems, total }) => (
    <CheckoutPageContainer>
        <CheckoutHeaderContainer>
            <HeaderBlockContainer>
                <span>Product</span>
            </HeaderBlockContainer>
            <HeaderBlockContainer>
                <span>Description</span>
            </HeaderBlockContainer>
            <HeaderBlockContainer>
                <span>Quantity</span>
            </HeaderBlockContainer>
            <HeaderBlockContainer>
                <span>Price</span>
            </HeaderBlockContainer>
            <HeaderBlockContainer>
                <span>Remove</span>
            </HeaderBlockContainer>
        </CheckoutHeaderContainer>
        {cartItems.map(cartItem=> <CheckoutItem key={cartItem.id}cartItem={cartItem}/>)}
        <TotalContainer>
            <span>TOTAL: ${total}</span>
        </TotalContainer>
        <TestWarningContainer>
            *Please use this test credit card for payments*
            <br/>
            4242 4242 4242 4242 - Exp: 01/23 - CVV: 123
        </TestWarningContainer>
        <StripeButton price={total} />
    </CheckoutPageContainer>
)

const mapStateToProps = createStructuredSelector({
    cartItems: cartItemsSelector,
    total: cartItemsTotalSelector
})

export default connect(mapStateToProps)(CheckoutPage);