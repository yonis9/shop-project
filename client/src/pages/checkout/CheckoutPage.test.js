import React from 'react';
import { shallow } from 'enzyme';

import { CheckoutPage } from './CheckoutPage';

// cartItems, total

describe('CheckoutPage component', () => {
    let wrapper;

    beforeEach(() => {
        const mockProps = {
            cartItems: [],
            total: 0
        };

        wrapper = shallow(<CheckoutPage {...mockProps}/>);
    });

    it('should render CheckoutPage component', () => {
        expect(wrapper).toMatchSnapshot();
    });
});