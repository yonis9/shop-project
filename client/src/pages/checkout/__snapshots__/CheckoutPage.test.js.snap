// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutPage component should render CheckoutPage component 1`] = `ShallowWrapper {}`;
